lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -899,35 +899,132 @@ class MongoClientInterface {
+     * with updating the operation log.
+     * @param {Object} log - The logger to use.
+     * @param {Function} cb - The callback function to call when the operation is complete. It is called with an error
+     * if there is an issue with the operation.
+     * @returns {Promise} A promise that resolves when the operation is complete. The promise is rejected with an error
+     * if there is an issue with the operation.
      */
-    putObjectNoVer(c, bucketName, objName, objVal, params, log, cb) {
-        const masterKey = formatMasterKey(objName, params.vFormat);
-        c.updateOne({
-            _id: masterKey,
-        }, {
+    putObjectNoVer(collection, bucketName, objName, value, params, log, cb) {
+        if (params?.needOplogUpdate) {
+            return this.putObjectNoVerWithOplogUpdate(collection, bucketName, objName, value, params, log, cb);
+        }
+        const key = formatMasterKey(objName, params.vFormat);
+        const putFilter = { _id: key };
+        return collection.updateOne(putFilter, {
             $set: {
-                _id: masterKey,
-                value: objVal,
+                _id: key,
+                value,
             },
         }, {
             upsert: true,
-        }).then(() => cb()).catch((err) => {
-            log.error('putObjectNoVer: error putting obect with no versioning', { error: err.message });
+        }).then(() => cb()).catch(err => {
+            log.error('internalPutObject: error putting object',
+                { bucket: bucketName, object: key, error: err.message });
             return cb(errors.InternalError);
         });
     }
 
+    /**
+     * Updates an object in a MongoDB collection without changing its version.
+     * If the object doesn't exist, it will be created (upsert is true for the second update operation).
+     * The operation is logged in the oplog.
+     *
+     * @param {Object} collection - The MongoDB collection to update the object in.
+     * @param {string} bucketName - The name of the bucket the object belongs to.
+     * @param {string} objName - The name of the object.
+     * @param {Object} value - The new value of the object.
+     * @param {Object} params - Additional parameters.
+     * @param {string} params.vFormat - object key format
+     * @param {string} params.originOp - origin operation
+     * @param {Object} log - The logger to use.
+     * @param {Function} cb - The callback function to call when the operation is complete.
+     * It is called with an error if there is an issue with the operation.
+     * @returns {void}
+     */
+    putObjectNoVerWithOplogUpdate(collection, bucketName, objName, value, params, log, cb) {
+        const key = formatMasterKey(objName, params.vFormat);
+        const putFilter = { _id: key };
+        // filter used when finding and updating object
+        const findFilter = Object.assign({
+            _id: key,
```

Comment:
redundant with `, putFilter` a few lines below...

also, whenever possible, prefer to use `...` to "unpack" object instead of Object.assign, as it will be easier to migrate to typescript


```suggestion
    const findFilter = {
        ...putFilter,
        $or: [
            { 'value.deleted': { $exists: false } },
            { 'value.deleted': { $eq: false } },
        ],
    };
```

Reply:
done here : https://github.com/scality/Arsenal/pull/2249/commits/043b64c3d17ad2bbcc3ea34ddc2b3f6590c436b1#diff-9c207befbd8f4d79a3b6e3d7d143444e804ca1684236b8fd9fa0f9e03c53477aR961-R971